### ISSUE TITLE
`runIrlbaSVD()` on DelayedMatrix

### ISSUE THREAD

**Issue by drisso** [LTLA/BiocSingular#1]
I'm trying to run 

```{r}
svd <- runIrlbaSVD(x)
```
where `x` is a DelayedMatrix, but I get the following error:

```{r}
Error in W[, j_w] <- avj : 
  number of items to replace is not a multiple of replacement length
```

This is the same error that I get if I run `irlba::irlba()` directly on `x`. Looking at the `runIrlbaSVD()` code, it looks like it's indeed passing a DelayedMatrix (the result of `standardize_matrix()`) to `irlba::irlba()`. 

Obviously, this works:
```{r}
svd <- runIrlbaSVD(as.matrix(x))
```

But is there a way to use irlba on DelayedMatrices or should I stick to random or exact SVD?

**Comment by LTLA** [comment]
This is a known issue, waiting for _irlba_ update to propagate bwlewis/irlba#45.

**Comment by drisso** [comment]
Thanks, @LTLA!

**Comment by LTLA** [comment]
This should be fixed in the latest _irlba_ (2.3.3, just came out today).
